distributed/core.py

```diff
@@ -39,6 +40,46 @@
 from . import protocol
 
 
+class Status(Enum):
+    """
+    This Enum contains the various states a worker, scheduler and nanny can be
+    in. Some of the status can only be observed in one of nanny, scheduler or
+    worker but we put them in the same Enum as they are compared with each
+    other.
+    """
+
+    closed = "closed"
+    closing = "closing"
+    closing_gracefully = "closing-gracefully"
+    init = "init"
+    running = "running"
+    starting = "starting"
+    stopped = "stopped"
+    stopping = "stopping"
+    undefined = None
```

Comment:
It looks like closing/closed and stopping/stopped might be redundant.  I would be in favor of unifying around one pair of values, presumably closing/closed.

Reply:
I don't have a sense for how often users check these values and if there is a way that we should make this change smoothly.  I'm inclined to make a small breaking change here, just because I don't think that many people depend on these values.

Reply:
There are many ways of handling this, in the properties you can emit a warning if you get a given value assigned tell users to use the Enum, and same in the `__eq__` method, if you see comparison with the deprecated values, you warn. 

Those warnings can be turned into error in tests. 

You can also help refactor with having two variants of the enum map to the same value.